cppForSwig/BlockDataMap.cpp

```diff
@@ -44,15 +44,24 @@ void BlockData::deserialize(const uint8_t* data, size_t size,
    {
       //light tx deserialization, just figure out the offset and size of
       //txins and txouts
-      vector<size_t> offsetIns, offsetOuts;
+      vector<size_t> offsetIns, offsetOuts, offsetsWitness;
```

Comment:
we don't want witness data for scanning purposes, this is supposed to light and fast. refer to the in code command.

BlockDataMap is a specific to scan operations. It avoid copies when possible and does not preemptively compute hashes unless asked for explictly